### ISSUE TITLE
[A11Y] Accordion - iOS Voiceover does not read expanded state

### ISSUE THREAD

**Issue by jamesshannon** [ampproject/amphtml#12259]
For a11y purposes the aria-expanded attribute is recommended. This causes voiceover (and other screenreaders) to state whether or not the item is currently expanded.

This is WAI in Chrome and Safari desktop but using the Accordion demo on AMP by Example iOS Safari does not say the expanded state of the accordion.

**Comment by aghassemi** [comment]
Interesting, thanks for raising this. We do put `aria-expanded`, will have to investigate to see if voice over has support for this or just a bug in voice-over we can workaround somehow.

**Comment by jamesshannon** [comment]
VoiceOver does have support for this. You can see it working on the red expandables on https://www.statefarm.com/simple-insights/auto-and-vehicles/industry-wide-rise-in-auto-insurance-rates (on both MacOS and iOS). It just doesn't work with amp-accordion. My suspicion is that the DOM structure is important (with aria-expanded) needing to be on an element at a particular level.

**Comment by robdodson** [comment]
My hunch is that the structure is slightly off. The ARIA Authoring Practices Guide 1.1 [Accordion pattern](https://www.w3.org/TR/wai-aria-practices-1.1/#accordion) suggests:

> Each accordion header **button** is wrapped in an element with role heading that has a value set for aria-level that is appropriate for the information architecture of the page.
(emphasis added to match emphasis in the spec)

Meaning there's an actual `<button>` inside of the heading.

> If the accordion panel associated with an accordion header is visible, the header button element has aria-expanded set to true. If the panel is not visible, aria-expanded is set to false.

So the structure you probably want is something like:

```html
<h2>
  <button id="option1" aria-expanded="true" aria-controls="region1">
    Option 1
  </button>
</h2>
<div id="region1" role="region" aria-labelledby="option1">
  Hello World
</div>
```

Although `aria-expanded` is [technically supported](https://www.w3.org/TR/wai-aria-1.1/#heading) on `role=heading` elements. It sounds like VoiceOver is just maybe not picking it up in this case.
